File: Chickito/src/main/java/com/ftn/Chickito/service/SectorService.java
package com.ftn.Chickito.service;

import com.ftn.Chickito.model.Sector;
import com.ftn.Chickito.model.enums.SectorType;

public interface SectorService {

    Sector findByCompanyAndType(Long companyId, SectorType type);
}

File: Chickito-front/src/app/modules/admin/services/sector.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from '../../core/services/api.service'; 
import { ConfigService } from '../../core/services/config.service'; 
import { _throw } from 'rxjs/observable/throw'; 
import {  map } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class SectorService {

  constructor(
    private apiService: ApiService, 
    private config: ConfigService,) {}

    findByCompanyAndType(companyId : any, sectorType: any) { 

      return this.apiService.get(this.config.sector_url + `/findByCompanyAndType/${companyId}/${sectorType}` )
      .pipe(map(sector => { 
        return sector;
      }));
    }
}
